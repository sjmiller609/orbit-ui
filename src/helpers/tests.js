import { BrowserRouter } from 'react-router-dom'
import { shape } from 'prop-types'
import { mount } from 'enzyme'

// Instantiate router context
const router = {
  history: new BrowserRouter().history,
  route: {
    location: {},
    match: {},
  },
}

const createContext = () => ({
  context: { router },
  childContextTypes: { router: shape({}) },
})

// wrap with router context - to mount RR links correctly
export function mountWrap(node) {
  return mount(node, createContext())
}
