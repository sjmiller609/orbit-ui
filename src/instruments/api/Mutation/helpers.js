const has = (str, find) => ~str.toLowerCase().indexOf(find.toLowerCase())

export const formErrors = ({ error, errors }) => {
  if (!error) return

  let err = JSON.stringify(error).toLowerCase()
  return errors.find(e => {
    if (e.key && has(err, e.key)) return e
    else if (e.allKeys && e.allKeys.every(k => has(err, k))) return e
    else if (e.someKeys && e.someKeys.some(k => has(err, k))) return e
  })
}
