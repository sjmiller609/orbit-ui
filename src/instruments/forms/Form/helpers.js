// faltten and stringify nested keys
export const unpack = (obj, ns, hash) => {
  if (!obj) return
  return Object.entries(obj).reduce((acc, [key, value]) => {
    const keypath = ns ? `${ns}.${key}` : key
    if (value && typeof value === 'object') {
      return unpack(value, keypath, acc)
    }
    acc[keypath] = value
    return acc
  }, hash || {})
}

const roll = (keys, obj, value) => {
  const key = keys[0]
  // case 1 - not obj
  if (keys.length === 1) {
    obj[key] = value
    return obj
  }
  // case 2 - doesn't exist
  if (!obj[key]) {
    obj[key] = isNaN(keys[1]) ? {} : []
  }
  // case 3 - exists
  obj[key] = roll(
    keys.slice(1),
    Array.isArray(obj[key]) ? [...obj[key]] : { ...obj[key] },
    value
  )

  return obj
}

// convert back into obj
export const pack = (obj, type = {}) => {
  return Object.entries(obj).reduce((acc, [key, value]) => {
    const keys = key.split('.')

    return roll(keys, acc, value)
  }, type)
}

export const packChild = ({ name, obj }) => {
  const obj2 = {}
  let isArray
  Object.entries(obj).forEach(k => {
    if (name.length < k[0].length && k[0].indexOf(name) === 0) {
      const n = k[0].slice(name.length + 1)
      // must be true for all matching
      isArray = !isNaN(n.slice(0, n.indexOf('.')))
      obj2[n] = k[1]
    }
  })
  if (Object.keys(obj2).length) return pack(obj2, isArray ? [] : {})
  return null
}

export const removeChild = ({ name, obj = {} }) => {
  const obj2 = {}
  Object.entries(obj).forEach(k => {
    if (name.length >= k[0].length || k[0].indexOf(name) !== 0) {
      obj2[k[0]] = k[1]
    }
  })
  return obj2
}
