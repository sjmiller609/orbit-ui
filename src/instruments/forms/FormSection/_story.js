import React from 'react'
import { storiesOf } from '@storybook/react'
import { FormSection } from 'instruments'

storiesOf('Instruments|Forms.FormSection', module).add('Default', () => (
  <FormSection
    title="Mock Form Section"
    text="Mock description for this section">
    [Replace with form parts]
  </FormSection>
))
