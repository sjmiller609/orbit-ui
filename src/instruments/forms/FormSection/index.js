'use strict'
import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'
import s from './styles.scss'
import { H5 } from 'instruments'

const FormSection = ({ title, id = 'section', children, className }) => {
  return (
    <section id={id} className={classnames(s.section, className)}>
      {title && <H5 className={s.title}>{title}</H5>}
      {children}
    </section>
  )
}

FormSection.propTypes = {
  title: PropTypes.string,
  children: PropTypes.oneOfType([PropTypes.array, PropTypes.element]),
  className: PropTypes.string,
  id: PropTypes.string,
}

export default FormSection
