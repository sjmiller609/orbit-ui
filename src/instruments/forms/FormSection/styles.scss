@import 'src/styles/mixins.scss';

.section {
  &:not(:first-child) {
    margin-top: 3rem;
  }
  &:not(:last-child) {
    margin-bottom: 3rem;
  }
  .title {
    padding-top: 2.2rem;
    &::after {
      display: block;
      background: linear-gradient(
        to right,
        rgba($lightGray, 0.5),
        $lightGray 10%,
        rgba($lightGray, 0.1)
      );
      content: '';
      position: relative;
      bottom: -0.9rem;
      width: 100%;
      height: 2px;
      left: 0;
      transform: scale(1);
      margin-bottom: 4rem;
    }
  }
}
