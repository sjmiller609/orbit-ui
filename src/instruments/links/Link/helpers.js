// check if url is external
export const externalUrl = url => {
  if (typeof url === 'object') return false
  const path = /^https?:\/\/|^\/\//i
  if (path.test(url) && !~url.indexOf(window.location.origin)) return true
  return
}
