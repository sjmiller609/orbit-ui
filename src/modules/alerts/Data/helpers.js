import { formErrors } from 'instruments'

const errors = [
  {
    key: 'permissions',
    name: 'properties.alert_emails',
    error: 'Contact your workspace admin to upgrade your permissions.',
  },
]

export const handleError = error => formErrors({ error, errors })

export const trimError = error => error.split(':')[1].trim()
