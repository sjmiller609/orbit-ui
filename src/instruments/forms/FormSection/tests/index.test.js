import React from 'react'
import { mountWrap } from 'helpers/tests'
import FormSection from '../index'

const props = {
  title: 'Test',
  text: 'Test section',
  children: ['<p>Test child</p>'],
  className: 'testClassString',
  id: '1234',
}

describe('FormSection', () => {
  it('renders correctly', () => {
    const wrapper = mountWrap(<FormSection {...props} />)
    expect(wrapper).toMatchSnapshot()
  })
})
