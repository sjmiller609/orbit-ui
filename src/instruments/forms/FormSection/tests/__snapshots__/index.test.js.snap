// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormSection renders correctly 1`] = `
<FormSection
  className="testClassString"
  id="1234"
  text="Test section"
  title="Test"
>
  <section
    className="section hasText testClassString"
    id="1234"
  >
    <H5
      className="title"
    >
      <h5
        className="h5 title"
      >
        Test
      </h5>
    </H5>
    <Mini
      className="text"
    >
      <p
        className="mini text"
      >
        Test section
      </p>
    </Mini>
    &lt;p&gt;Test child&lt;/p&gt;
  </section>
</FormSection>
`;
